docs/source/advanced/sagemaker-pipelines.rst

```diff
@@ -0,0 +1,303 @@
+* ``--region``: AWS region. (Required)
+* ``--graphstorm-pytorch-cpu-image-uri``: GraphStorm GConstruct/dist_part/train/inference CPU ECR image URI. (Required)
+* ``--graphstorm-pytorch-gpu-image-uri``: GraphStorm GConstruct/dist_part/train/inference GPU ECR image URI.
+* ``--gsprocessing-pyspark-image-uri``: GSProcessing SageMaker PySpark ECR image URI. (Required if running a ``gsprocessing`` job.)
+
+Instance Configuration
+^^^^^^^^^^^^^^^^^^^^^^
+
+* ``--instance-count`` / ``--num-parts``: Number of worker instances/partitions for partition, training, inference. (Required)
+* ``--cpu-instance-type``: CPU instance type. (Default: ml.m5.4xlarge)
+* ``--gpu-instance-type``: GPU instance type. (Default: ml.g5.4xlarge)
+* ``--train-on-cpu``: Run training and inference on CPU instances instead of GPU. (Flag)
+* ``--graph-construction-instance-type``: Instance type for graph construction.
+* ``--gsprocessing-instance-count``: Number of GSProcessing instances (PySpark cluster size, default is equal to ``--instance-count``).
+* ``--volume-size-gb``: Additional volume size for SageMaker instances in GB. (Default: 100)
+
+Task Configuration
+^^^^^^^^^^^^^^^^^^
+
+* ``--graph-name``: Name of the graph. (Required)
+* ``--input-data-s3``: S3 path to the input graph data. (Required)
+* ``--output-prefix-s3``: S3 prefix for the output data. (Required)
+* ``--pipeline-name``: Name for the pipeline.
+* ``--base-job-name``: Base job name for SageMaker jobs. (Default: 'gs')
+* ``--jobs-to-run``: Space-separated string of jobs to run in the pipeline.
+  Possible values are: ``gconstruct``, ``gsprocessing``, ``dist_part``, ``gb_convert``, ``train``, ``inference`` (Required).
+* ``--log-level``: Logging level for the jobs. (Default: INFO)
+* ``--step-cache-expiration``: Expiration time for the step cache. (Default: 30d)
+* ``--update-pipeline``: Update an existing pipeline instead of creating a new one. (Flag)
+
+Graph Construction Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* ``--graph-construction-config-filename``: Filename for the graph construction config.
+* ``--graph-construction-args``: Additional parameters to be passed directly to the GConstruct/GSProcessing job.
+
+Partition Configuration
+^^^^^^^^^^^^^^^^^^^^^^^
+
+* ``--partition-algorithm``: Partitioning algorithm to use. (Default: random)
+* ``--partition-input-json``: Name for the JSON file that describes the input data for distributed partitioning. (Default: updated_row_counts_metadata.json)
+* ``--partition-output-json``: Name for the output JSON file that describes the partitioned data generated by GConstruct or GSPartition.
+  (Default: metadata.json for GSPartition,  use <graph_name>.json for ``gconstruct``.)
+
+Training Configuration
+^^^^^^^^^^^^^^^^^^^^^^
+
+* ``--model-output-path``: S3 path for model output.
+* ``--num-trainers``: Number of trainers (per-instance training processes) to use during training/inference. Set this equal to number of GPUs (Default: 4)
```

Comment:
The default GPU instance type only has 1 GPUs, so if a user sets up his pipeline with all default arguments, the training task will fail.